docs/reference/how-to/avoid-oversharding.asciidoc

```diff
@@ -0,0 +1,73 @@
+[[avoiding-oversharding]]
+== Avoid oversharding
+
+Sometimes reducing the number of shards in a cluster while maintaining the same amount of data will lead to a more effective use of system resources (CPU, RAM, IO). In these situations, we consider the cluster “oversharded”. The number of shards where this inflection point occurs depends on a variety of factors, including available hardware, indexing load, data volume, the types of queries executed against the cluster, the rate of these queries being issued, and the volume of data being queried. Testing against production data with production queries on production hardware is the only way to calibrate optimal shard sizes. Shard sizes of tens of GB are commonly used, and this may be a useful starting point from which to experiment. Kibana’s {kibana-ref}/elasticsearch-metrics.html[Elasticsearch monitoring] provides a useful view of historical cluster performance when evaluating the impact of different shard sizes.
+
+[float]
+=== Why oversharding is inefficient
```

Comment:
Where possible, we try to add explicit anchors. It ensure that links to the content are less likely to break.

We also try to use `[discrete]` over `[float]` as that's preferred in Asciidoctor. You'll still see lots of historical `[float]` attributes tho.
```suggestion
[discrete]
[[oversharding-inefficient]]
=== Why oversharding is inefficient
```